exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Player.java

```diff
@@ -0,0 +1,57 @@
+package com.adaptionsoft.games.uglytrivia;
+
+public class Player {
+    private static final int MAX_TILES = 12;
+    private final String playerName;
+    private int currentTile;
+    private int coins;
+    private boolean isInPenaltyBox;
+
+    public Player (String playerName) {
+        this.playerName = playerName;
+        this.currentTile = 0;
+        this.coins = 0;
+        this.isInPenaltyBox = false;
+    }
+
+    public String getPlayerName() {
+        return playerName;
+    }
+
+
+    public int getCurrentTile() {
+        return currentTile;
+    }
+
+    public int getCoins() {
+        return coins;
+    }
+
+    public boolean isInPenaltyBox() {
+        return isInPenaltyBox;
+    }
+
+    public void putInPenaltyBox() {
+        this.isInPenaltyBox = true;
+    }
+
+    void playerMoves(int roll) {
+        currentTile = currentTile + roll;
+        if (currentTile > 11) currentTile = currentTile - MAX_TILES;
+
+        System.out.println(playerName
+                + "'s new location is "
+                + currentTile);
+    }
+
+    public void addCoins (int coins) {
```

Comment:
Great way of extracting a Player data and functionality,  maybe coins methods  could be stored  in the the Game class as it is not strictly just Player method but rather is influenced buy other factors in a game.

Reply:
Thank you! My thought process was that the player should keep track of the players coins rather than the game, and also it solves issues that would arise with the private coins variable within Player. There are arguments for both sides, but I'll have a go at implementing your suggestions and see what it comes out as.